Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -1340,87 +1347,102 @@ namespace AzToolsFramework
+                if (parentEntityId.IsValid() && (entitiesThatWillBeRemoved.find(parentEntityId) == entitiesThatWillBeRemoved.end()))
+                {
+                    parentEntitysThatNeedUpdate.insert(GetEntityById(parentEntityId));
+                }
 
-                    removedEntityDomAndPathList.emplace_back(
-                        entityDomInOwningTemplatePath.Get(commonOwningTemplateDom), AZStd::move(entityAliasPath));
+                AZStd::unique_ptr<Instance> detachedInstance = commonOwningInstance->get().DetachNestedInstance(instanceAlias);
 
-                    entitiesThatGotRemoved.insert(entityId);
+                if (isOverrideEditing)
+                {
+                    detachedInstanceAliasPaths.push_back(AZStd::move(PrefabDomUtils::PathStartingWithInstances + instanceAlias));
+                }
+                else
+                {
+                    // Removes the link if it is source template editing.
+                    RemoveLink(detachedInstance, commonOwningInstance->get().GetTemplateId(), undoBatch.GetUndoBatch());
                 }
 
-                PrefabUndoHelpers::RemoveEntities(removedEntityDomAndPathList, commonOwningTemplateId, undoBatch.GetUndoBatch());
+                detachedInstance.reset();
             }
 
-            // Step 3 - Updates parent entity DOMs with new children information.
-            for (const auto& [parentEntityId, parentEntityDomBefore] : parentEntityDomBeforeRemovalMap)
+            // 2. Detaches entity objects.
+            for (const AZ::Entity* entity : entityList)
             {
-                // Skips if the parent entity was already deleted.
-                if (entitiesThatGotRemoved.find(parentEntityId) != entitiesThatGotRemoved.end())
+                const AZ::EntityId entityId = entity->GetId();
+                const AZStd::string nestedEntityAliasPath = m_instanceToTemplateInterface->GenerateEntityAliasPath(entityId);
+
+                // Captures the parent entity that needs to be updated.
+                AZ::EntityId parentEntityId;
+                AZ::TransformBus::EventResult(parentEntityId, entityId, &AZ::TransformBus::Events::GetParentId);
+
+                if (parentEntityId.IsValid() && (entitiesThatWillBeRemoved.find(parentEntityId) == entitiesThatWillBeRemoved.end()))
                 {
-                    continue;
+                    parentEntitysThatNeedUpdate.insert(GetEntityById(parentEntityId));
                 }
 
-                const AZ::Entity* parentEntity = GetEntityById(parentEntityId);
-                if (parentEntity)
-                {
-                    PrefabDom parentEntityDomAfter;
-                    m_instanceToTemplateInterface->GenerateDomForEntity(parentEntityDomAfter, *parentEntity);
+                commonOwningInstance->get().DetachEntity(entityId).release();
+                AZ::ComponentApplicationBus::Broadcast(&AZ::ComponentApplicationRequests::DeleteEntity, entityId);
```

Comment:
These are not new changes but it is interesting that to delete entities, we are calling this ebus but to delete entities in nested prefabs, we are simply calling detachedInstance.reset();. This is not consistent behavior for entities and prefabs. We should simply be calling reset() in both cases here. Can you give that a try and see if that works? If it breaks anything, please add a ghi to track this.

Reply:
I tried it and it seemed to work with no issue.

I think the difference between the two ways is ComponentApplication::DeleteEntity also signals "entity removed event". Code Ref:

https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp#L1171

But I did not notice any "entity removed event" handlers being registered for this. I also did a code search and are not seeing DetachEntity(entitiyId).reset() usage in other places.

Reply:
It is a good thing to be worrying about backwards compatibility. However, there is also the consistency issue where we are only calling this api for direct entities only
. And since there doesn't appear to be other systems that are using this additional event, I think it's a safe thing to start calling reset() now
.

If you want to separate it from your PR, you can try it in a follow up PR so that it can be easily reverted if needed

Reply:
Discussed offline. The inconsistency is between deleting direct entities and deleting entities in nested instances (during detachedInstance.reset()). The first one involves calling the DeleteEntity event API.

I will put up a follow-up PR for this change.